video/out/vo_gpu_next.c

```diff
@@ -1461,18 +1487,25 @@ static int control(struct vo *vo, uint32_t request, void *data)
         return true;
     }
 
-    case VOCTRL_HDR_METADATA: {
-        struct mp_hdr_metadata *hdr = data;
-        hdr->min_luma = p->last_hdr_metadata.min_luma;
-        hdr->max_luma = p->last_hdr_metadata.max_luma;
-        hdr->max_cll = p->last_hdr_metadata.max_cll;
-        hdr->max_fall = p->last_hdr_metadata.max_fall;
-        hdr->scene_max[0] = p->last_hdr_metadata.scene_max[0];
-        hdr->scene_max[1] = p->last_hdr_metadata.scene_max[1];
-        hdr->scene_max[2] = p->last_hdr_metadata.scene_max[2];
-        hdr->scene_avg = p->last_hdr_metadata.scene_avg;
-        hdr->max_pq_y = p->last_hdr_metadata.max_pq_y;
-        hdr->avg_pq_y = p->last_hdr_metadata.avg_pq_y;
+    case VOCTRL_HDR_METADATA:
+        copy_hdr_metadata_to_mp(&p->last_hdr_metadata, data);
+        return true;
+
+    case VOCTRL_IMAGE_OUT_PARAMS: {
```

Comment:
Do you need this? I confess I didn't look very closely but can't you just get the frame from the vo and then get the image from that?

Reply:
The vo_frame is actually the input to render, what I want to get is the target parameters of the render. It doesn't fit into vo_frame.

In properties terms
`video-target-params` -> target display params
`video-out-params` -> video out params. In fact it is "video in" (video out after filter chain), it is the input frame that VO holds and renders
`video-dec-params` -> video params after decode

I could hold something like target params in `struct vo`, but I think `VOCTRL` is more suited here.

As always, good ideas welcome. I don't know every inch of mpv :)

Reply:
Oh hmm, I can't say I have any better ideas. This is indeed information only the specific VO would know.